spec/21-MQTT3_FORMAT.md

```diff
@@ -0,0 +1,51 @@
+# MQTT 3 recommendation
```

Comment:
Comment from @zteo-google

In general I think it's not a good idea to modify a user payload to carry the trace context. This can cause breakages in parsers and would be considered a disruptive feature. We should treat the user payload as a black box that should not be tampered with.

My recommendation is that traces be carried on a completely separate MQTT topic, preferably one that is considered 'system reserved' in MQTT (for eg. topics $SYS/traces or something similar). For packet correlation purposes, either the packet ID (QoS 1 and 2), or a digest of the original message can be used.

Reply:
@zteo-google this section is mentioned as non-normative. Meaning it's just a recommendation for how it can be done if one decided to do it

Reply:
```suggestion
# MQTT v3 recommendation
```